posthog/temporal/batch_exports/transformer.py

```diff
@@ -225,4 +301,37 @@ def write_batch(self, record_batch: pa.RecordBatch) -> typing.Generator[bytes, N
         yield data
 
 
+def transform_record_batch_from_shared_memory(shm_name: str, transformer: StreamTransformer):
+    """Top level function to run transformers in multiprocessing.
+
+    Record batches are accessed from shared memory.
+    """
+    shm = None
+    record_batch = None
+    buffer_reader = None
+    reader = None
+
+    try:
+        shm = sm.SharedMemory(name=shm_name)
+        buffer_reader = pa.BufferReader(shm.buf)
+        with pa.RecordBatchStreamReader(buffer_reader) as reader:
+            record_batch = reader.read_next_batch()
```

Comment:
there should only ever be a single batch right?

Reply:
I did try a fixed worker model but didn't notice any improvements beyond the new baseline this PR brings. And the queues added a lot of complexity overhead.